src/sys/windows/selector.rs

```diff
@@ -274,6 +272,21 @@ impl SockState {
     }
 }
 
+/// Converts the pointer to a `SockState` into a raw pointer.
+/// To revert see `from_overlapped`.
+fn into_overlapped(sock_state: Pin<Arc<Mutex<SockState>>>) -> PVOID {
+    let overlapped_ptr: *const Mutex<SockState> =
+        unsafe { Arc::into_raw(Pin::into_inner_unchecked(sock_state)) };
```

Comment:
This breaks CI tests on Windows. If this gets merged as it is, it will block testing on Windows side because CI will always fail here. 

error[E0658]: use of unstable library feature 'pin_into_inner'
   --> src\sys\windows\selector.rs:279:32
    |
279 |         unsafe { Arc::into_raw(Pin::into_inner_unchecked(sock_state)) };
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: for more information, see https://github.com/rust-lang/rust/issues/60245

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `mio`.

To learn more, run the command again with --verbose.